src/crypto/tls/tls_cf_test.go

```diff
@@ -17,3 +19,49 @@ func TestPropagateCFControl(t *testing.T) {
 		t.Errorf("failed to propagate CFControl: got %v; want %v", got, want)
 	}
 }
+
+// If the client uses the wrong KEM algorithm to offer ECH, the ECH provider
+// should reject rather than abort. We check for this condition by looking at
+// the error returned by hpke.Reciver.Setup(). This test asserts that the
```

Comment:
```suggestion
// the error returned by hpke.Receiver.Setup(). This test asserts that the
```

Reply:
And perhaps the comment could explicitly point to the `EXP_ECHKeySet.GetDecryptionContext`?

Alternatively, how feasible would a unit test be for that function?